index.js

```diff
@@ -11,17 +11,18 @@ function init({ algoliaConfig, sitemapLocation, hitToParams }) {
   const index = client.initIndex(algoliaConfig.indexName);
   const sitemaps = [];
 
-  const handleSitemap = entries => {
-    const iterator = sitemaps.length;
-    const sitemap = createSitemap(entries);
-    saveSiteMap({ sitemap, index: iterator, root: sitemapLocation.path });
+  const handleSitemap = async entries => {
     sitemaps.push({
-      loc: `${sitemapLocation.href}/sitemap.${iterator}.xml`,
+      loc: `${sitemapLocation.href}/${await saveSiteMap({
```

Comment:
either this could be moved outside the same call and saved in a variable, or the arrow function could be a inline function instead of a block